File: v131016/TeeNew/Surface_Smooth.pas
unit Surface_Smooth;
{$I TeeDefs.inc}

interface

uses
  {$IFNDEF LINUX}
  Windows, Messages,
  {$ENDIF}
  SysUtils, Classes,
  {$IFDEF CLX}
  QGraphics, QControls, QForms, QDialogs, QExtCtrls, QStdCtrls, QComCtrls,
  {$ELSE}
  Graphics, Controls, Forms, Dialogs, ExtCtrls, StdCtrls, ComCtrls,
  {$ENDIF}
  Base, TeEngine, TeeSurfa, TeeProcs, Chart;

type
  TSurfaceSmooth = class(TBaseForm)
    Series1: TSurfaceSeries;
    CheckBox1: TCheckBox;
    procedure FormCreate(Sender: TObject);
    procedure CheckBox1Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

implementation

{$IFNDEF CLX}
{$R *.DFM}
{$ELSE}
{$R *.xfm}
{$ENDIF}

procedure TSurfaceSmooth.FormCreate(Sender: TObject);
begin
  inherited;
  Series1.FillSampleValues(10);
end;

procedure TSurfaceSmooth.CheckBox1Click(Sender: TObject);
begin
  inherited;
  Series1.SmoothPalette:=CheckBox1.Checked
end;

initialization
  RegisterClass(TSurfaceSmooth);
end.

File: TeeNew/Builder2010/Surface_Smooth.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Surface_Smooth.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Base"
#pragma link "TeeSurfa"
#pragma resource "*.dfm"
TSurfaceSmooth *SurfaceSmooth;
//---------------------------------------------------------------------------
__fastcall TSurfaceSmooth::TSurfaceSmooth(TComponent* Owner)
        : TBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TSurfaceSmooth::FormCreate(TObject *Sender)
{
  Series1->FillSampleValues(10);
}
//---------------------------------------------------------------------------

void __fastcall TSurfaceSmooth::CheckBox1Click(TObject *Sender)
{
  Series1->SmoothPalette = CheckBox1->Checked;
}
//---------------------------------------------------------------------------

